lib/rules/newline-after-var.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @fileoverview Rule to check empty newline after "var" statement
+ * @author Gopal Venkatesan <http://g13n.me>
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+    var mode = context.options[0],
+        validator;              // regex to validate the rule
+
+    if (mode === "always") {
+        validator = /\S(?:\r?\n){2,}\S?/;
+    } else if (mode === "never") {
+        validator = /\S(?:\r?\n)?\S/;
+    }
+
+    return {
+        "VariableDeclaration:exit": function(node) {
+            var lastToken = context.getLastToken(node),
+                nextToken = context.getTokenAfter(node),
+                // peek few characters beyond the last token (typically the semi-colon)
+                sourceLines = context.getSource(lastToken, 0, 3);
+
+            // Some coding styles like Google uses multiple `var` statements.
+            // So if the next token is a `var` statement don't do anything.
+            if (nextToken && nextToken.type === "Keyword" && nextToken.value === "var") {
```

Comment:
I think what @nzakas is saying is what if the next statement is a `let` or a `const`.

Some of the use case to test for:

```.js
{ code: "var greet = 'hello';\n let name = 'world'; console.log(greet);", args: [2, "foobar"] },
{ code: "var greet = 'hello';\n const name = 'world'; console.log(greet);", args: [2, "foobar"] },
{ code: "let greet = 'hello';\n var name = 'world'; console.log(greet);", args: [2, "foobar"] },
{ code: "let greet = 'hello';\n const name = 'world'; console.log(greet);", args: [2, "foobar"] }
```